Automattic/jetpack
# Add Tracks events for Welcome Flow

## Proposed changes:
We're adding the following Tracks events to cover interactions in the Welcome Flow together with recommendations section:

Welcome Banner:
- Banner Dismiss: `jetpack_myjetpack_welcome_banner_dismiss_click`
- Step View: `jetpack_myjetpack_welcome_banner_step_view`
  - Props: `{ current_step: 'connection' | 'evaluation' }`
- Survey Submit: `jetpack_myjetpack_welcome_banner_evaluation_submit`
  - Props:`{ goals: ( 'protect' | 'performance' | 'audience' | 'content' | 'unsure' )[] }`

Recommendation section:
- Recommendations View: `jetpack_myjetpack_evaluation_recommendations_view`
  - Props: `{ modules: JetpackModule }`
- Recommendations Redo: `jetpack_myjetpack_evaluation_recommendations_redo_click`
- Recommendations Dismiss: `jetpack_myjetpack_evaluation_recommendations_dismiss_click`
- Recommendations Purchase Click:`jetpack_myjetpack_evaluation_recommendations_checkout_click`
  - Props: `{ slug: JetpackModule }`

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Run JN site with Jetpack pointing this branch: https://jurassic.ninja/create?jetpack-beta&branches.jetpack=add/welcome-flow-tracks-events
* Notice the described Tracks events while going through the flow (you can use Tracks Vigilante)
